### PR TITLE
WIPSymbol Token

### PR DESCRIPTION
**Pull request by byronlin-bq**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ion/binaryreader.go [hunk 1]
```diff
@@ -27,6 +27,18 @@ func (r *binaryReader) SymbolTable() SymbolTable {
 	return r.lst
 }
 
+func (r *binaryReader) FieldNameSymbol() (*SymbolToken, error) {
+	if r.fieldNameSid == UnknownSid {
+		return NewSymbolToken(nil, UnknownSid, nil), nil
+	}
+	text, ok := r.SymbolTable().FindByID(uint64(r.fieldNameSid))
+
+	if ok != true {
```
Comment (matthewh-BQ) [inline ion/binaryreader.go:36]: ok is a boolean, this can be converted into !ok.

File: ion/symboltable.go [hunk 2]
```diff
@@ -113,6 +114,22 @@ func (s *sst) Adjust(maxID uint64) SharedSymbolTable {
 	}
 }
 
+func (s *sst) Find(text string) (SymbolToken, bool) {
+	emptyString := ""
+
+	id, ok := s.FindByName(text)
+	if ok != true {
```
Comment (matthewh-BQ) [inline ion/symboltable.go:121]: Can be refactored to !ok.

File: ion/symboltable.go [hunk 3]
```diff
@@ -113,6 +114,22 @@ func (s *sst) Adjust(maxID uint64) SharedSymbolTable {
...
+		return *NewSymbolToken(&emptyString, UnknownSid, nil), false
+	}
+
+	internedText, ok := s.FindByID(id)
+	if ok != true {
```
Comment (matthewh-BQ) [inline ion/symboltable.go:126]: Can be refactored to !ok.

File: ion/textreader.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ func (s trs) String() string {
 	}
 }
 
-// A textReader is a Reader that reads text Ion.
+// A textReader is a Reader that reads text Iongo build.
```
Comment (matthewh-BQ) [inline ion/textreader.go:39]: Looks like a typo that's carried over from trying to run `go build`.

File: ion/symboltable.go [hunk 5]
```diff
@@ -283,6 +304,28 @@ func (t *lst) MaxID() uint64 {
...
+	}
+
+	emptyString := ""
+	id, ok := t.FindByName(s)
+	if ok != true {
+		return *NewSymbolToken(&emptyString, UnknownSid, nil), false
+	}
+
+	internedText, ok := t.FindByID(id)
+	if ok != true {
```
Comment (matthewh-BQ) [inline ion/symboltable.go:322]: Can be refactored to !ok.

File: ion/symboltable.go [hunk 6]
```diff
@@ -283,6 +304,28 @@ func (t *lst) MaxID() uint64 {
...
+	for _, imp := range t.imports {
+		symbolToken, ok := imp.Find(s)
+		if ok {
+			return *NewSymbolToken(symbolToken.text, UnknownSid, nil), true
+		}
+	}
+
+	emptyString := ""
+	id, ok := t.FindByName(s)
+	if ok != true {
```
Comment (matthewh-BQ) [inline ion/symboltable.go:317]: Can be refactored to !ok.

File: ion/binaryreader.go (hunk 1 above)
Comment (byronlin-bq) [reply ion/binaryreader.go]: Addressed

File: ion/symboltable.go (hunk 2 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Addressed

File: ion/symboltable.go (hunk 3 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Addressed

File: ion/textreader.go (hunk 4 above)
Comment (byronlin-bq) [reply ion/textreader.go]: Reverted

File: ion/symboltable.go (hunk 5 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Addressed

File: ion/symboltable.go (hunk 6 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Addressed

**Pull request closed without merging**